lib/NormalModule.js

```diff
@@ -69,260 +42,457 @@ function contextify(options, request) {
+			if(this.preventParsing(noParseRule, this.request)) {
+				return callback();
+			}
+
+			try {
+				this.parser.parse(this._source.source(), {
+					current: this,
+					module: this,
+					compilation: compilation,
+					options: options
+				});
+			} catch(e) {
+				const source = this._source.source();
+				const error = new ModuleParseError(this, source, e);
+				this.markModuleAsErrored(error);
+				return callback();
+			}
+			return callback();
+		});
+	}
+
+	getHashDigest() {
+		const hash = crypto.createHash("md5");
+		this.updateHash(hash);
+		return hash.digest("hex");
 	}
-	var _source = this._source;
-	if(!_source) return new RawSource("throw new Error('No source available');");
-	var source = new ReplaceSource(_source);
-	this._cachedSource = {
-		source: source,
-		hash: hash
-	};
-	var topLevelBlock = this;
-
-	function doDep(dep) {
-		var template = dependencyTemplates.get(dep.constructor);
-		if(!template) throw new Error("No template for dependency: " + dep.constructor.name);
-		template.apply(dep, source, outputOptions, requestShortener, dependencyTemplates);
+
+	sourceDependency(dependency, dependencyTemplates, source, outputOptions, requestShortener) {
+		const template = dependencyTemplates.get(dependency.constructor);
+		if(!template) throw new Error("No template for dependency: " + dependency.constructor.name);
+		template.apply(dependency, source, outputOptions, requestShortener, dependencyTemplates);
 	}
 
-	function doVariable(availableVars, vars, variable) {
-		var name = variable.name;
-		var expr = variable.expressionSource(dependencyTemplates, outputOptions, requestShortener);
+	sourceVariables(variable, availableVars, dependencyTemplates, outputOptions, requestShortener) {
+		const name = variable.name;
+		const expr = variable.expressionSource(dependencyTemplates, outputOptions, requestShortener);
 
-		function isEqual(v) {
-			return v.name === name && v.expression.source() === expr.source();
+		if(availableVars.some(v => v.name === name && v.expression.source() === expr.source())) {
+			return;
 		}
-		if(availableVars.some(isEqual)) return;
-		vars.push({
+		return {
 			name: name,
 			expression: expr
+		};
+	}
+
+	variableInjectionFunctionWrapperStartCode(varNames) {
+		const openingIIFEParanthesis = "(";
+		const args = varNames.join(", ");
+		return `/* WEBPACK VAR INJECTION */${openingIIFEParanthesis}function(${args}) {`;
+	}
+
+	contextArgument(block) {
+		if(this === block) {
+			return this.exportsArgument || "exports";
+		}
+		return "this";
+	}
+	variableInjectionFunctionWrapperEndCode(varExpressions, block) {
+		const firstParam = this.contextArgument(block);
+		const furtherParams = varExpressions.map(e => e.source()).join(", ");
+		const closingIIFEParanthesis = ")";
+		return `}.call(${firstParam}, ${furtherParams})${closingIIFEParanthesis}`;
+	}
+
+	splitVariablesInUniqueNamedChunks(vars) {
+		const chunkCollection = vars.reduce((chunkCol, variable) => {
```

Comment:
This can be simplied to not require a tempChunk.

``` js
return vars.reduce((chunks, variable) => {
  const current = chunks[chunks.length - 1];
  if(current.some(v => v.name === variable.name)) {
    chunks.push([variable]);
  } else {
    current.push(variable);
  }
  return chunks;
}, [[]]);
```